azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -184,6 +186,17 @@ def update(self):
         if self._extensions_goal_state is None or most_recent.created_on_timestamp > self._extensions_goal_state.created_on_timestamp:
             self._extensions_goal_state = most_recent
 
+        # For Fast Track goal states, verify that the required certificates are in the goal state
+        if self.extensions_goal_state.source == GoalStateSource.FastTrack:
+            for extension in self.extensions_goal_state.extensions:
+                for settings in extension.settings:
+                    if settings.protectedSettings is None:
+                        continue
+                    certificates = self.certs.summary
+                    if not any(settings.certificateThumbprint == c['thumbprint'] for c in certificates):
+                        message = "Certificate {0} needed by {1} is missing from the goal state".format(settings.certificateThumbprint, extension.name)
```

Comment:
In this case, do we skip GS or just log error and keep going?

Reply:
At this point it just writes to the local log to, should this ever happen, make the error obvious, and to collect telemetry.

As far as handling the error, at this point the code already handles all the scenarios we anticipated. If we are missing other scenarios the telemetry would point this out and then we can decide how to handle those.